### PR TITLE
Client should track current broker leader

### PR DESCRIPTION
**Pull request by otoolep**

### CODE REVIEW

**Comment by otoolep** [comment]
@benbjohnson 

This may not be a complete implementation, but should be an improvement.

**Comment by benbjohnson** [comment]
Looks good, can you add a test?

**Comment by otoolep** [comment]
Cool, let me look into a test now.

**Comment by otoolep** [comment]
@benbjohnson -- can you take another look? I added tests, and also fixed a bug -- the *entire* redirect URL was getting set in the client, when it should only have been the host (and port if present).

File: messaging/client.go [hunk 1]
```diff
@@ -83,9 +84,30 @@ func (c *Client) LeaderURL() *url.URL {
...
+// SetLeaderURL sets the explicit broker leader. The leader is set to the scheme, host,
+// and port (if any) contained in the URL. All other components of the URL are ignored.
+func (c *Client) SetLeaderURL(u *url.URL) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if u == nil {
+		return
+	}
+
+	v, err := url.Parse(u.Scheme + "://" + u.Host)
```
Comment (benbjohnson) [inline messaging/client.go]: It's way easier to just create a new `url.URL`:

```
v := &url.URL{Scheme: u.Scheme, Host: u.Host}
```

**Comment by benbjohnson** [comment]
One comment on the URL copy but otherwise lgtm.

File: messaging/client.go (hunk 1 above)
Comment (otoolep) [inline messaging/client.go]: Got it -- I felt there had to be a better way.

**Comment by otoolep** [comment]
Thanks for the feedback @benbjohnson -- made that one final change. I will merge on green.

**Comment by otoolep** [comment]
Green build, merging now.

**Pull request merged**